Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift

```diff
@@ -165,15 +189,269 @@ struct IDEDiagnosticConsoleFormatter: DiagnosticConsoleFormatter {
+        }
+        
+        // A range containing the source lines and some surrounding context.
+        let sourceRange = Range(
+            uncheckedBounds: (
+                lower: max(1, diagnosticRange.lowerBound.line - Self.contextSize) - 1,
+                upper: min(sourceLines.count, diagnosticRange.upperBound.line + Self.contextSize)
+            )
+        )
+        let maxLinePrefixWidth = String(sourceRange.upperBound).count
+        
+        var suggestionsPerLocation = [SourceLocation: [String]]()
+        for solution in solutions {
+            // Solutions that requires multiple or zero replacements
+            // will be shown at the beginning of the diagnostic range. 
+            let location: SourceLocation
+            if solution.replacements.count == 1 {
+                location = solution.replacements.first!.range.lowerBound
+            } else {
+                location = diagnosticRange.lowerBound
+            }
+
+            suggestionsPerLocation[location, default: []].append(solution.summary)
+        }
+
+        result.append("\n\(String(repeating: " ", count: maxLinePrefixWidth))--> ")
+        result.append(        "\(formattedSourcePath(url)):\(diagnosticRange.lowerBound.line):\(diagnosticRange.lowerBound.column)-\(diagnosticRange.upperBound.line):\(diagnosticRange.upperBound.column)"
+        )
+
+        for (sourceLineIndex, sourceLine) in sourceLines[sourceRange].enumerated() {
+            let lineNumber = sourceLineIndex + sourceRange.lowerBound + 1
+            let linePrefix = "\(lineNumber)".padding(toLength: maxLinePrefixWidth, withPad: " ", startingAt: 0)
+
+            let highlightedSource = highlightSource(
+                sourceLine: sourceLine,
+                lineNumber: lineNumber,
+                range: diagnosticRange
+            )
+            
+            let separator: String
+            if lineNumber >= diagnosticRange.lowerBound.line && lineNumber <= diagnosticRange.upperBound.line {
+                separator = "+"
+            } else {
+                separator = "|"
+            }
+
+            result.append("\n\(linePrefix) \(separator) \(highlightedSource)")
+
+            var columnsWithSuggestions = Set<Int>()
+            var suggestionsPerColumn = [(Int, [String])]()
+
+            for (location, suggestions) in suggestionsPerLocation {
+                if location.line == lineNumber {
+                    suggestionsPerColumn.append((location.column, suggestions))
+                    columnsWithSuggestions.insert(location.column)
+                }
+            }
+
+            let suggestionLinePrefix = String(repeating: " ", count: maxLinePrefixWidth) + " |"
+
+            for (columnNumber, columnSuggestions) in suggestionsPerColumn.sorted(by: { $0.0 > $1.0 }) {
+                var prefix = suggestionLinePrefix
+
+                for column in 0...columnNumber - 1 {
```

Comment:
When I first looked at this loop I didn't realize that it was creating a string with multiple lines like
```
  |  |       | 
```
It would be good with a comment that explain what this does and how this prefix it used.

----

It would also be possible to compute this prefix once per line and then get the relevant substring for each column. E.g. something like:

```swift
let sortedColumns = suggestionsPerColumn.keys.sorted(by: >)

var longestPrefix = [Character](repeating: " ", count: sortedColumns.first! + 1)
for column in sortedColumns {
    longestPrefix[column] = "│"
}

for columnNumber in sortedColumns {
    let prefix = suggestionLinePrefix + String(longestPrefix.prefix(columnNumber))
    // ...
}
```